File: Template/Template.Model/StudentRepository.cs
using NakedObjects;
using System.Linq;


namespace Template.Model
{
    public class StudentRepository
    {
        #region Injected Services
        //An implementation of this interface is injected automatically by the framework
        public IDomainObjectContainer Container { set; protected get; }
        #endregion
        public Student CreateNewStudent()
        {
            return Container.NewTransientInstance<Student>();
        }

        public IQueryable<Student> AllStudents()
        {
            return Container.Instances<Student>();
        }

        public IQueryable<Student> FindStudentByName(string name)
        {
            return AllStudents().Where(c => c.FullName.ToUpper().Contains(name.ToUpper()));
        }
    }

}

File: Template/NOF2 Server/Template.Model/StudentRepository.vb
Namespace Services
    Public Class StudentRepository
        Implements IContainerAware

        Public Property Container As IContainer Implements IContainerAware.Container

        Public Function ActionAllStudents() As IQueryable(Of Student)
            Return Container.AllInstances(Of Student)
        End Function

        Public Function ActionFindStudents(match As String) As IQueryable(Of Student)
            Return Container.AllInstances(Of Student)
        End Function

    End Class
End Namespace
